src/code/z_debug.c

```diff
@@ -1,20 +1,327 @@
 #include <ultra64.h>
 #include <global.h>
+#include <z64.h>
+#include <color.h>
+#include <regs.h>
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug/func_800636C0.s")
+#define BUTTON_C_RIGHT        0x0001
+#define BUTTON_C_LEFT         0x0002
+#define BUTTON_C_DOWN         0x0004
+#define BUTTON_C_UP           0x0008
+#define BUTTON_R              0x0010
+#define BUTTON_L              0x0020
+#define BUTTON_D_RIGHT        0x0100
+#define BUTTON_D_LEFT         0x0200
+#define BUTTON_D_DOWN         0x0400
+#define BUTTON_D_UP           0x0800
+#define BUTTON_START          0x1000
+#define BUTTON_Z              0x2000
+#define BUTTON_B              0x4000
+#define BUTTON_A              0x8000
 
-void func_8006375C(UNK_TYPE arg0, UNK_TYPE arg1, UNK_TYPE arg2)
+#define REG_GROUP_TOTAL 29
+#define REG_N 96
+#define PAGE_N 16
+
+typedef struct
+{
+    u8 x;
+    u8 y;
+    u8 colorId;
+    char text[0x15];
+} PrintTextBuffer;
+
+typedef struct
 {
+    u16 push;
+    u16 held;
+} InputCombo;
+
+GameInfo*       gGameInfo;
+int             D_8015FA94; //no known symbols
+PrintTextBuffer D_8015FA98[0x16];
+
+s16 D_8011E0B0 = 0; //PrintTextBuffer index
+Color_RGBA8 printTextColors[] = {
+    { 0xFF, 0xFF, 0x20, 0xC0 },
+    { 0xFF, 0x96, 0x80, 0xC0 },
+    { 0x80, 0x60, 0x00, 0x40 },
+    { 0xC0, 0x80, 0x10, 0x80 },
+    { 0xFF, 0xC0, 0x20, 0x80 },
+    { 0xE6, 0xE6, 0xDC, 0x40 },
+    { 0x80, 0x96, 0xFF, 0x80 },
+    { 0x80, 0xFF, 0x20, 0x80 },
+};
+
+InputCombo inputCombos[REG_GROUP_TOTAL] = {
+    { BUTTON_L, BUTTON_C_UP },
+    { BUTTON_L, BUTTON_C_LEFT },
+    { BUTTON_L, BUTTON_C_DOWN },
+    { BUTTON_L, BUTTON_A },
+    { BUTTON_R, BUTTON_C_DOWN },
+    { BUTTON_L, BUTTON_C_RIGHT },
+    { BUTTON_L, BUTTON_R },
+    { BUTTON_L, BUTTON_D_LEFT },
+    { BUTTON_L, BUTTON_D_RIGHT },
+    { BUTTON_L, BUTTON_D_UP },
+    { BUTTON_L, BUTTON_B },
+    { BUTTON_L, BUTTON_Z },
+    { BUTTON_L, BUTTON_D_DOWN },
+    { BUTTON_R, BUTTON_A },
+    { BUTTON_R, BUTTON_B },
+    { BUTTON_R, BUTTON_Z },
+    { BUTTON_R, BUTTON_L },
+    { BUTTON_R, BUTTON_C_UP },
+    { BUTTON_R, BUTTON_C_RIGHT },
+    { BUTTON_R, BUTTON_D_LEFT },
+    { BUTTON_R, BUTTON_C_LEFT },
+    { BUTTON_R, BUTTON_START },
+    { BUTTON_L, BUTTON_START },
+    { BUTTON_R, BUTTON_D_RIGHT },
+    { BUTTON_R, BUTTON_D_UP },
+    { BUTTON_START, BUTTON_R },
+    { BUTTON_START, BUTTON_A },
+    { BUTTON_START, BUTTON_B },
+    { BUTTON_START, BUTTON_C_RIGHT },
+};
+
+char regChar[] = " SOPQMYDUIZCNKXcsiWAVHGmnBdkb";
+
+//initialize GameInfo
+void func_800636C0()
+{
+    s32 i;
+
+    gGameInfo = (GameInfo*)SystemArena_MallocDebug(sizeof(GameInfo), "../z_debug.c", 0x104);
+    gGameInfo->reg_page = 0;
+    gGameInfo->reg_group = 0;
+    gGameInfo->reg_cur = 0;
+    gGameInfo->dpad_last = 0;
+    gGameInfo->repeat = 0;
+    for (i = 0; i < 0xAE0; i++)
```

Comment:
I would use ARRAY_COUNT here if it works
```suggestion
    for (i = 0; i < ARRAY_COUNT(gGameInfo->data); i++)
```